backend/src/routes/auth.ts

```diff
@@ -0,0 +1,70 @@
+import express from 'express'
+import { body, validationResult, CustomValidator } from 'express-validator'
+import {User, UserSchema} from '../database/models/user'
+import * as bcrypt from 'bcrypt'
+import * as jwt from 'jsonwebtoken'
+import * as fs from 'fs'
+
+const auth = express.Router();
+
+const isValidRole: CustomValidator = value => {
+    return value === "teacher" || value === "student";
+  };
+
+const privateKey = fs.readFileSync('resources/private.key');
+const publicKey = fs.readFileSync('resources/public.key');
+
+auth.post("/register",
+          body('email').isEmail(),
+          body('password').isLength({ min: 8 }),
+          body('role').custom(isValidRole),
+          async (req, res) => {
+            const errors = validationResult(req);
+            if (!errors.isEmpty()) return res.status(400).json({ errors: errors.array() });
```

Comment:
```
if (!validationResult(req).isEmpty())
```
zamiast zmiennych

Reply:
okej, ale potem zwracam errors.array() - wydawało mi się że jest to też trochę bardziej estetyczne bo linijki są krótsze xd

Reply:
okej w tym przypadku ma to sens zeby 2 razy tej samej funkcji nie wolac